plugins/joke.lua

```diff
@@ -0,0 +1,27 @@
+-- !joke, fetch jokes from icndb
+
+local http_request = require "http.request"
+local json = require "dkjson"
+
+return {
+	hooks = {
+		PRIVMSG = function(irc, state, sender, origin, message, pm) -- luacheck: ignore 212
+			if not message:match("^!joke") then
+				return
+			end
+			local first = message:match("^!joke%s+(%g+)")
+			if not first then
+				first = sender[1]
+			end
+			local h, s = assert(http_request.new_from_uri(
```

Comment:
Currently these callbacks don't have an error handler: any error thrown will kill the whole bot.
i.e. using `assert` on a remote resource isn't advised!